src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketTeamMetadataAction.java

```diff
@@ -72,20 +79,66 @@ public BitbucketAvatarCacheSource(String serverUrl, StandardCredentials credenti
         }
 
         @Override
-        public AvatarImage fetch() {
-            BitbucketAuthenticator authenticator = AuthenticationTokens
-                    .convert(BitbucketAuthenticator.authenticationContext(serverUrl), credentials);
-            BitbucketApi bitbucket = BitbucketApiFactory.newInstance(serverUrl, authenticator, repoOwner, null);
+        public AvatarImage fetch(StandardCredentials credentials) {
             try {
-                return bitbucket.getTeamAvatar();
+                return doFetch(credentials);
             } catch (IOException e) {
+                if (e.getCause() instanceof BitbucketRequestException) {
+                    BitbucketRequestException bre = (BitbucketRequestException) e.getCause();
+                    if (bre.getHttpCode()==401) {
+                        // credentials not updated here maybe we need to refresh them
+                        StandardCredentials standardCredentials = findCredentials();
+                        // try again with refreshed credentials
+                        // TODO compare previous and new token if we really need to try again
+                        if (standardCredentials != null) {
+                            this.credentials = standardCredentials;
+                            try {
+                                return doFetch(this.credentials);
+                            } catch (IOException | InterruptedException ex) {
+                                LOGGER.log(Level.INFO, ex.getClass().getName()+": " + e.getMessage(), e);
+                            }
+                        }
+                    }
+                }
                 LOGGER.log(Level.INFO, "IOException: " + e.getMessage(), e);
             } catch (InterruptedException e) {
                 LOGGER.log(Level.INFO, "InterruptedException: " + e.getMessage(), e);
             }
             return null;
         }
 
+        private StandardCredentials findCredentials() {
+            try (ACLContext ctx = ACL.as(ACL.SYSTEM)) {
+                return CredentialsMatchers.firstOrNull(
+                        CredentialsProvider.lookupCredentials(
+                                credentials.getClass(),
+                                (Item) null, // context
```

Comment:
That would only found root global credentials and not folder credentials. If the credentials are defined in a folder or in the Multibranch item, this is unlikely to work ?
Don't we have access to the SCMOwner from here ?

Reply:
not when this piece of code wants to retrieve the avatar https://github.com/jenkinsci/branch-api-plugin/blob/7332a9019a0109975bfa537fa00a6f98e74e3721/src/main/java/jenkins/branch/MetadataActionFolderIcon.java#L84

Reply:
yup agree it's a possible problem..
but not sure how to fix it without changing the branch-api-plugin as well.

Reply:
I have a better solution but this need changes in scm-api and branch-api as well.
not sure we want this in the same PR

Reply:
would this be a regression or edge casewith the new implementation?

Reply:
this PR fix issue with the current implementation at least for system credentials.
fixing it for folder as well will need more work, I will start the work for that.
But at least we could merge and release this PR as it's already better than the current code ;) 
wdyt?

Reply:
yeah and if I understand correctly, if folder credentials are used, the lookup will not find it and no fetch will be done. So worst case scenario, we don't have the avatar anymore, but no fetch, so no account locked down in bitbucket.
Good enough IMO.